doc/metrics.md

```diff
@@ -0,0 +1,80 @@
+
+### ETCD metrics
+
+The metrics under the `etcd` prefix/namespace are carried forward from etcd library that we use. These metrics does not include details of the `etcd` deployment on which `etcd-backup-restore` utility works. Instead, it helps the monitoring the `embedded etcd` we spawn while restoration process.
+
+
+### Snapshot
+
+These metrics describe the status of the snapshotter. In order to detect outages or problems for troubleshooting, these metrics should be closely monitored. The below mentioned metrics are listed as collection of series using prometheus labels `kind` and `succeeded`. `Kind` label indicates the snapshot kind i.e. full snapshot or incremental/delta snapshot in the context. And succeeded indicates whether the metrics is for successful operation or erroneous operation.
+
+| Name | Description | Type |
+|------|-------------|------|
+| etcdbr_snapshot_duration_seconds | Total latency distribution of saving snapshot to object store. | Histogram |
+| etcdbr_snapshot_gc_total | Total number of garbage collected snapshots. | Counter |
+| etcdbr_snapshot_latest_revision | Revision number of latest snapshot taken. | Gauge |
+| etcdbr_snapshot_latest_timestamp | Timestamp of latest snapshot taken. | Gauge |
+
+Abnormally high snapshot duration (`etcdbr_snapshot_duration_seconds`) indicates disk issues and low network bandwidth.
+
+`etcdbr_snapshot_latest_timestamp` indicates the time last snapshot was taken. If it is too older than current time, then it indicates either the snapshots are being skipped because of no updates on etcd or :warning: something fishy is going on and you might be loosing large data in case of next restoration.
+
+`etcdbr_snapshot_gc_total` gives the total number of snapshots garbage collected since bootstrap. You can use this in coordination with `etcdbr_snapshot_duration_seconds_count` to get number of snapshots in object store.
+
+### Defragmentation
+
+The metrics for defragmentation is of type histogram, which would gives count the number of times defragmentation triggered. :warning: The deframentation latency should be low as possible, since
+defragmentation is indeed costly operation, which results in unavailability of etcd for the period.
+
+| Name | Description | Type |
+|------|-------------|------|
+| etcdbr_defragmentation_duration_seconds | Total latency distribution of defragmentation of etcd data directory. | Histogram |
+
+### Validation and Restoration
+
+Two major steps in initialization of etcd data directory are validation and restoration. Its needed to monitor the count of there calls and time duration from high availability perspective.
+
+| Name | Description | Type |
+|------|-------------|------|
+| etcdbr_validation_duration_seconds | Total latency distribution of validating data directory. | Histogram |
+| etcdbr_restoration_duration_seconds | Total latency distribution of restoring from snapshot. | Histogram |
+
```

Comment:
Would it be better to have `SubSystems` for defragmentation, validation and restoration as well? In case of any new metrics being added in the future. Just a thought.

Reply:
This is first implementation, over the period with further iteration we might have to change it.  Unlike, snapshot subsystem, I don't have clear vision about what extra metrics will be there related to restoration/validation/defragmentation. So, let's keep it flat for now.  Converting flat to hierarchical model, will be easy comparative to changing one hierarchical to another hierarchical structure.